### ISSUE TITLE
Gogs need a static website hosting feature.

### ISSUE THREAD

**Issue by nmabhinandan** [gogits/gogs#1314]
It would be great to have a static site hosting feature similar to github pages. There is already a static site generator written in go. 
http://github.com/spf13/hugo

**Comment by windsource** [comment]
+1

**Comment by morph027** [comment]
+1 ;)

**Comment by Unknwon** [comment]
So... how would the URL be look like?

`https://try.gogs.io/pages/hi.html`?

**Comment by morph027** [comment]
Sure, why not. Any reverse proxy can take care of other DNS targets...

**Comment by bzar** [comment]
+1, currently doing this with git hooks, but it's practically an admin-only feature because of security concerns.

**Comment by j4** [comment]
+1

**Comment by hdhog** [comment]
+1

**Comment by natir** [comment]
Currently I work to solve this issue, you can see the code https://github.com/natir/gogs/tree/develop.

The url look like https://try.gogs.io/username/reponame/pages/hi.html

**Comment by bkcsoft** [comment]
@natir There is a _really_ good reason why github puts gh-pages on `*.github.io` instead of `*.github.com` though... you should really read up on that before continuing :smile: 

http://thenextweb.com/insider/2013/04/07/github-moves-pages-from-domain-over-security-concerns/#gref
TL;DR; https://news.ycombinator.com/item?id=9624246

I solve gh-pages with webhooks, CD and nginx :smile:

**Comment by bkcsoft** [comment]
@tboerger WRT https://github.com/go-gitea/gitea/issues/43#issuecomment-110610813 yes macaron has [multi-host](https://go-macaron.com/docs/middlewares/switcher) support :)

**Comment by elouin** [comment]
+1
Any Progress?

**Comment by Unknwon** [comment]
Reference: https://pages.gitlab.io/

**Comment by fronbasal** [comment]
Any news?

**Comment by filipemateus** [comment]
+1

**Comment by chaosbunker** [comment]
+1

any news?

**Comment by liuyatao** [comment]
+1

**Comment by fronbasal** [comment]
Anything new?

**Comment by duckbrain** [comment]
Perhaps this would be better done with a separate application. [Drone](https://github.com/drone/drone) is a CI that integrates with Gogs by logging in with user credentials and adding webhooks to be notified of changes.

I don't think Gogs should have this functionality because to do it securely, it should have a different domain.  (And not break sites that don't use relative paths) Some may want to host from a different server or use something other than hugo, etc.

That said, Gogs could have a better API for these sorts of things. Websockets could be a good way to receive notifications of repository changes, and system level API could prevent needing a user's credentials being used in applications that connect to Gogs, but those would be separate issues.

**Comment by rumkin** [comment]
I think it's better to create list of default undeletable hooks. User can see them, but could not remove. Why is it matter: there is many tools that should to send feedback to repository's owner when it fails. The only issue is that it could contain some secret, which should not be shown to repository owner.

**Comment by ShalokShalom** [comment]
With a proper Caddy setup is this probably easy to implement: https://caddyserver.com/docs/http.git

**Comment by denzuko** [comment]
TLDR; Wow... Such github.. much duplication.. just use drone, docker images, and webpack-markdown-loader.

Why not just have drone do the work via ghost/jekyll/webpack then have the static site bundled up into a docker image and deployed from there?

Sound convoluted maybe but behind the scenes that's _all_ github is doing for gh pages.

I could see a feature that gogs adds to the repo `.drone.yml` and `.drone/` (much like the README, Licence, gitignore) for building the image, pushing it to docker hub, and executing either of the tree static site generators listed above.
